Firebase/InAppMessagingDisplay/Card/FIDCardViewController.m

```diff
@@ -0,0 +1,137 @@
+                resourceBundle);
+    return nil;
+  }
+  FIDCardViewController *cardVC =
+      (FIDCardViewController *)[storyboard instantiateViewControllerWithIdentifier:@"card-view-vc"];
+  cardVC.displayDelegate = displayDelegate;
+  cardVC.cardDisplayMessage = cardMessage;
+  cardVC.timeFetcher = timeFetcher;
+
+  return cardVC;
+}
+
+- (IBAction)primaryActionButtonTapped:(id)sender {
+  if (self.cardDisplayMessage.primaryActionURL) {
+    FIRInAppMessagingAction *primaryAction = [[FIRInAppMessagingAction alloc]
+        initWithActionText:self.cardDisplayMessage.primaryActionButton.buttonText
+                 actionURL:self.cardDisplayMessage.primaryActionURL];
+    [self followAction:primaryAction];
+  } else {
+    [self dismissView:FIRInAppMessagingDismissTypeUserTapClose];
+  }
+}
+
+- (IBAction)secondaryActionButtonTapped:(id)sender {
+  if (self.cardDisplayMessage.secondaryActionURL) {
+    FIRInAppMessagingAction *secondaryAction = [[FIRInAppMessagingAction alloc]
+        initWithActionText:self.cardDisplayMessage.secondaryActionButton.buttonText
+                 actionURL:self.cardDisplayMessage.secondaryActionURL];
+    [self followAction:secondaryAction];
+  } else {
+    [self dismissView:FIRInAppMessagingDismissTypeUserTapClose];
+  }
+}
+
+- (FIRInAppMessagingDisplayMessage *)inAppMessage {
+  return self.cardDisplayMessage;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  self.cardView.backgroundColor = self.cardDisplayMessage.displayBackgroundColor;
+
+  self.bodyTextView.contentInset = UIEdgeInsetsZero;
+  self.bodyTextView.textContainer.lineFragmentPadding = 0;
+
+  // make the background half transparent
+  [self.view setBackgroundColor:[UIColor.grayColor colorWithAlphaComponent:0.5]];
+
+  self.titleLabel.text = self.cardDisplayMessage.title;
+  self.titleLabel.textColor = self.cardDisplayMessage.textColor;
+
+  self.bodyTextView.text = self.cardDisplayMessage.body;
+
+  [self.primaryActionButton setTitle:self.cardDisplayMessage.primaryActionButton.buttonText
+                            forState:UIControlStateNormal];
+  [self.primaryActionButton
+      setTitleColor:self.cardDisplayMessage.primaryActionButton.buttonTextColor
+           forState:UIControlStateNormal];
+
+  if (self.cardDisplayMessage.secondaryActionButton) {
+    self.secondaryActionButton.hidden = NO;
+    [self.secondaryActionButton setTitle:self.cardDisplayMessage.secondaryActionButton.buttonText
+                                forState:UIControlStateNormal];
+    [self.secondaryActionButton
+        setTitleColor:self.cardDisplayMessage.secondaryActionButton.buttonTextColor
+             forState:UIControlStateNormal];
+  }
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+
+  if (self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassRegular ||
+      self.traitCollection.verticalSizeClass == UIUserInterfaceSizeClassCompact) {
+    NSData *imageData = self.cardDisplayMessage.landscapeImageData
```

Comment:
Nit: a comment explaining that landscape image is optional and it defaults to portrait would be good.